### PR TITLE
feat(rule_engine): default timeout for jq/2 and jq/3 with timeout

### PR DESCRIPTION
**Pull request by kjellwinblad**
This PR adds a default timeout of 10 seconds to the rule engine's `jq/2` function, and adds a new function `jq/3` (where the last parameter is a timeout value). The default timeout can be configured with the setting "`rule_engine.jq_function_default_timeout`".

Having a timeout when executing jq code in the rule engine is important as jq code can potentially run forever. Also, the Erlang jq library limits the number of jq programs that can execute concurrently so a jq program that loops forever could potentially also prevent a "non-buggy" jq program from ever starting.

I'm not sure that `rule_engine.jq_function_default_timeout` is the right place for the setting so please let me know if there is a better place. Also, I would like help to translate the documentation strings to Chinese.

This PR is just a proposal for how to add timeout to the jq function so we can still discuss alternative ways of implementing this.

### CODE REVIEW

**Review by zmstone — CHANGES_REQUESTED** [review]

File: apps/emqx_rule_engine/src/emqx_rule_engine_schema.erl [hunk 1]
```diff
@@ -218,3 +239,20 @@ validate_sql(Sql) ->
...
+        _ -> T
+    end.
+
+-spec to_duration_ms(Input) -> {ok, integer()} | {error, Input} when
+    Input :: string() | binary().
+to_duration_ms(Str) ->
+    case hocon_postprocess:duration(Str) of
+        I when is_number(I) -> {ok, ceiling(I)};
+        _ -> {error, Str}
+    end.
```
[the remark this answers is not in the document]
Comment (kjellwinblad) [reply apps/emqx_rule_engine/src/emqx_rule_engine_schema.erl]: Thanks, fixed [402ab7e](https://github.com/emqx/emqx/pull/7975/commits/402ab7e759f89956116a9a0e713879c75a7bdcc2)

File: apps/emqx_rule_engine/src/emqx_rule_engine_schema.erl [hunk 2]
```diff
@@ -19,8 +19,16 @@
 -include_lib("typerefl/include/types.hrl").
 -include_lib("hocon/include/hoconsc.hrl").
 
+-type duration_ms() :: integer().
+
+-typerefl_from_string({duration_ms/0, emqx_schema, to_duration_ms}).
```
[the remark this answers is not in the document]
Comment (kjellwinblad) [reply apps/emqx_rule_engine/src/emqx_rule_engine_schema.erl]: Thanks, fixed [402ab7e](https://github.com/emqx/emqx/pull/7975/commits/402ab7e759f89956116a9a0e713879c75a7bdcc2)

File: apps/emqx_rule_engine/i18n/emqx_rule_engine_schema.conf [hunk 3]
```diff
@@ -239,6 +239,17 @@ of the rule, then the string "undefined" is used.
                           }
                   }
 
+    rule_engine_jq_function_default_timeout {
+      desc {
+          en: "Default timeout for the jq/2 rule engine function"
```
Comment (zmstone) [inline apps/emqx_rule_engine/i18n/emqx_rule_engine_schema.conf:244]:
```suggestion
          en: "Default timeout for the `jq` rule engine function"
```

File: apps/emqx_rule_engine/i18n/emqx_rule_engine_schema.conf [hunk 4]
```diff
@@ -239,6 +239,17 @@ of the rule, then the string "undefined" is used.
...
+          zh: "Default timeout for the jq/2 rule engine function"
```
Comment (zmstone) [inline apps/emqx_rule_engine/i18n/emqx_rule_engine_schema.conf:245]:
```suggestion
          zh: “规则引擎内建函数 `jq` 默认时间限制”
```

File: apps/emqx_rule_engine/i18n/emqx_rule_engine_schema.conf [hunk 5]
```diff
@@ -239,6 +239,17 @@ of the rule, then the string "undefined" is used.
...
+      }
+      label: {
+          en: "Rule engine jq function default timeout"
+          zh: "Rule engine jq function default timeout"
```
Comment (zmstone) [inline apps/emqx_rule_engine/i18n/emqx_rule_engine_schema.conf:249]:
```suggestion
          zh: "规则引擎 jq 函数时间限制"
```

**Comment by terry-xiaoyu** [comment]
I copied this PR to https://github.com/emqx/emqx/pull/8006 and then pushed my commits to fix the CIs.

Close this one.

**Pull request closed without merging**